pandas/core/internals/construction.py

```diff
@@ -289,6 +291,15 @@ def ndarray_to_mgr(
         if values.ndim == 1:
             values = values.reshape(-1, 1)
 
+    elif (
+        using_copy_on_write()
+        and isinstance(values, np.ndarray)
+        and (dtype is None or is_dtype_equal(values.dtype, dtype))
+        and copy_on_sanitize
+    ):
+        values = np.array(values, order="F", copy=copy_on_sanitize)
```

Comment:
is the order="F" thing something we should be doing in general in cases with copy=True?

Reply:
We are operating on the transposed array when copying a DataFrame object, so not needed in that case

Reply:
xref https://github.com/pandas-dev/pandas/pull/44871 took a look at preserving order when doing copy. there are some tradeoffs here